src/unitxt/metrics.py

```diff
@@ -1387,6 +1392,98 @@ def compute(
         return self.pipe(inputs, batch_size=self.batch_size)
 
 
+class LlamaIndexCorrectnessMetric(BulkInstanceMetric):
+    """Custom metric class for evaluating correctness using Llama Index.
+
```

Comment:
It requires OpenAI API .  How is it specified?

Reply:
Using an API key in an environmental variable.

BTW, this is something we need to be careful about (or even consider having) as proprietary data can be leaked this way to OPENAI

Generally, this is a major issue with some packages like llama-index as they sometimes use OPENAI based models as default.

Reply:
I think the 

class LlamaIndexCorrectnessMetric(BulkInstanceMetric):

should br sn abstract class with method get_llm() .

Then there should be a class 

class LlamaIndexCorrectnessMetricByOpenAI(LlamaIndexCorrectnessMetric):

which would receive a parameter (model name, could be with default "gpt-3.5-turbo")
 
 and the get_llm method of it would be
 
  return OpenAI(mode_name)

Reply:
Then the name in the catalog would also be

correctness_by_lamma_index_openai

Then it will be clear which metric is used, and that it's with open.

Reply:
Also, I assume this should be the name of the returned score.